source/common/http/http1/codec_impl.cc

```diff
@@ -788,8 +788,9 @@ Envoy::StatusOr<int> ConnectionImpl::onHeadersCompleteBase() {
 
   header_parsing_state_ = HeaderParsingState::Done;
 
-  // Returning 2 informs http_parser to not expect a body or further data on this connection.
-  return handling_upgrade_ ? 2 : statusor.value();
+  // Returning HttpParserCode::NoBodyData informs http_parser to not expect a body or further data
```

Comment:
The various callbacks are inconsistent. Looking at it again, I feel like `HttpParserCode` should be the return type all up to `setAndCheckCallbackStatus(Or)` which should return the converted `int` so that we enforce that anywhere we could be returning an error code goes through those setters.

(I think this will mean onHeadersComplete/onHeadersCompleteBase need to change)

Reply:
@KBaichoo What are your thoughts about introducing an enum class that wraps HttpParserCode in order to provide type safety.

Possible example:
enum class ParseStatus {
  Success = HttpParserCode::Success,
  NoBody = HttpParserCode::NoBody,
  NoBodyData = HttpParserCode::NoBodyData,
}

And use this type to replace most int return types in the codec implementation.  The enumToSignedInt calls should happen in the functions defined in the http_parser_settings ConnectionImpl::settings_ block.

Reply:
That makes sense, I'll fix that as well in this PR.